src/main/java/pl/byczazagroda/trackexpensesappbackend/service/WalletServiceImpl.java

```diff
@@ -2,12 +2,35 @@
 
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
+import org.springframework.validation.annotation.Validated;
+import pl.byczazagroda.trackexpensesappbackend.dto.CreateWalletDTO;
+import pl.byczazagroda.trackexpensesappbackend.dto.WalletDTO;
+import pl.byczazagroda.trackexpensesappbackend.exception.WalletNotSavedException;
+import pl.byczazagroda.trackexpensesappbackend.mapper.WalletModelMapper;
+import pl.byczazagroda.trackexpensesappbackend.model.Wallet;
 import pl.byczazagroda.trackexpensesappbackend.repository.WalletRepository;
 
+import javax.validation.Valid;
+
 @Service
+@Validated
 @RequiredArgsConstructor
 public class WalletServiceImpl implements WalletService {
 
     private final WalletRepository walletRepository;
+    private final WalletModelMapper walletModelMapper;
+
+
+    @Override
+    public WalletDTO createWallet(@Valid CreateWalletDTO createWalletDTO) {
```

Comment:
Do we need to repeat @Valid annotations as we already put it in controller method signature?
If so then please, change also the method signature in WalletService interface.

Reply:
We need annotation `@Valid` on interface for sure. Then we will have it as well on implementation code.
We need as well annotation `@Valid` on Controller. But annotations from Controller and Service are separate. It's another layer of application.